cli_classic.c

```diff
@@ -555,6 +561,126 @@ int main(int argc, char *argv[])
 		goto out_shutdown;
 	}
 
+	if (lock){
+
+		unsigned char cmd[1] = {0x05};
+		unsigned char sr1, sr2;
+
+		msg_gdbg("1. Read SR1 and SR2\n");
+
+		//Read status register 1
+		cmd[0] = 0x05;
+		spi_send_command(fill_flash, sizeof(cmd), sizeof(sr1), cmd, &sr1);
+		msg_gdbg("SR1 = 0x%02x\n", sr1);
+
+		programmer_delay(100);
+
+		//Read status register 2
+		cmd[0] = 0x35;
+		spi_send_command(fill_flash, sizeof(cmd), sizeof(sr2), cmd, &sr2);
+		msg_gdbg("SR2 = 0x%02x\n", sr2);
+
+		if ((sr1 & 0x80) && (sr2 & 0x01)){
+			msg_ginfo("SPI flash is in One Time Program mode. Can't write to it anymore!\n");
+		}
+
+		/*
+			 Set SPI flash block protect to WP_RO region (0x200000 - 0x7FFFFF)
+		*/
+
+		msg_gdbg("2. Write new values to SR1 = 0x34, SR2 = 0x40\n");
+
+		programmer_delay(100);
+
+		// Write enable command
+		cmd[0] = 0x06;
+		spi_send_command(fill_flash, sizeof(cmd), 0, cmd, NULL);
+
+		programmer_delay(1000);
+
+		// Write status register 1
+		sr1 &= 0x83;		// 
+		sr1 |= 0x34;		// Set SEC, TB, BP2, BP1, BP0 bits
+		sr2 &= 0x41;
+		sr2 |= 0x40;		// Enable CMP to reach WP_RO region (0x200000 - 0x7FFFFF)
+		unsigned char cmd_write[3] = {0x01, sr1, sr2};
+		spi_send_command(fill_flash, sizeof(cmd_write), 0, cmd_write, NULL);
+
+		programmer_delay(1000);
+
+		msg_gdbg("3. Read updated SR1 and SR2\n");
+		//Verify if SR1 and SR2 are written correctly
+		//Read status register 1
+		cmd[0] = 0x05;
+		spi_send_command(fill_flash, sizeof(cmd), sizeof(sr1), cmd, &sr1);
+		msg_gdbg("SR1 = 0x%02x\n", sr1);
+
+		programmer_delay(100);
+
+		//Read status register 2
+		cmd[0] = 0x35;
+		spi_send_command(fill_flash, sizeof(cmd), sizeof(sr2), cmd, &sr2);
+		msg_gdbg("SR2 = 0x%02x\n", sr2);
+
+		programmer_delay(100);
+
+		if ((sr1 & 0x34) && (sr2 & 0x40)){
+			msg_ginfo("Block protection successfully set for WP_RO region!\n");
+		}
+		else
+		{
+			msg_ginfo("Set block protection for WP_RO region FAILED! Please retry --lock command\n");
```

Comment:
@piotr-kleins what if block protection is not set but you proceed with OTP mode? You get a bad lock... You cannot proceed when block protection is not set. You have to do `goto out_shutdown;` here.